### PR TITLE
[WIP] ES6 modules

### PR DESCRIPTION
**Pull request by koops76**
Do not merge.

### CODE REVIEW

**Comment by koops76** [comment]
ping @RReverser @cowlicks @Hainish.

File: chromium/.eslintrc.json [hunk 1]
```diff
@@ -4,21 +4,21 @@
     "es6": true,
     "webextensions": true
   },
+  "overrides": [
+    {
+      "files": ["background.js", "entry.js", "incognito.js", "rules.js" ,"store.js", "util.js"],
```
Comment (RReverser) [inline chromium/.eslintrc.json]: What's this needed for / why not `module` by default?

**Comment by RReverser** [comment]
Code looks good. Does module syntax work in browser extensions?

**Comment by koops76** [comment]
@RReverser No.

File: chromium/.eslintrc.json (hunk 1 above)
Comment (koops76) [reply chromium/.eslintrc.json]: `module` disables `/* exported */`

**Comment by cowlicks** [comment]
Is this compatible with node?

**Comment by koops76** [comment]
No, it is currently only compatible with Webpack or Rollup.

**Comment by koops76** [comment]
@Hainish Need help with integrating Rollup with the build system.

**Comment by RReverser** [comment]
@koops76 That should be pretty trivial - just add Rollup config and call it as part of makecrx and makexpi.

**Comment by koops76** [comment]
I'm not sure if Node is available in `makecrx.sh` and `makexpi.sh`.

**Comment by RReverser** [comment]
It is AFAIK, see @Hainish's comment in https://github.com/EFForg/https-everywhere/pull/12493#issue-256083902

**Comment by cowlicks** [comment]
I'm still skeptical about our need for Rollup. 

So it might be better to have a discussion about why we need rollup before implementing it. What advantages does it provide? What are the disadvantages?

**Comment by koops76** [comment]
@cowlicks Rollup is basically a very clever `cat *.js > out.js`. It concatenates the scripts and links them together, automatically dealing with name conflicts. It uses ES6 modules feature to understand how the scripts should be linked. It doesn't minify or obfuscate the produced code.

**Comment by cowlicks** [comment]
@koops76 is there a way with rollup/webpack to load the extension unpacked? I like to develop like this because changes are immediately reflected in the browser. It seems like it would break this.

**Comment by koops76** [comment]
@cowlicks Use watch mode in Rollup or Webpack.

**Pull request closed without merging**